src/core/prerender/utils.ts

```diff
@@ -34,12 +33,17 @@ export function extractLinks(
 
   // Extract from any <TAG href=""> to crawl
   if (crawlLinks) {
-    _links.push(
-      ...[...html.matchAll(LINK_REGEX)]
-        .map((m) => escapeHtml(m[1]))
-        .filter((m) => !decodeURIComponent(m).startsWith("#"))
-        .filter((link) => allowedExtensions.has(getExtension(link)))
-    );
+    walkSync(parseHTML(html), (node) => {
```

Comment:
Not sure about perf tradeoffs but as saw first example in ultrahtml readme is using async iteratable, maybe worth we also try async version?